GigeVision.Core/Services/Camera.cs

```diff
@@ -449,22 +447,19 @@ public async Task<bool> SetOffsetAsync(uint offsetX, uint offsetY)
             {
                 await Gvcp.TakeControl().ConfigureAwait(false);
             }
-            string[] registers = new string[2];
-            registers[0] = Gvcp.RegistersDictionary[nameof(RegisterName.OffsetX)];
-            registers[1] = Gvcp.RegistersDictionary[nameof(RegisterName.OffsetY)];
-            uint[] valueToWrite = new uint[] { offsetX, offsetY };
-            bool status = (await Gvcp.WriteRegisterAsync(registers, valueToWrite).ConfigureAwait(false)).Status == GvcpStatus.GEV_STATUS_SUCCESS;
-            GvcpReply reply = await Gvcp.ReadRegisterAsync(registers).ConfigureAwait(false);
-            if (reply.Status == GvcpStatus.GEV_STATUS_SUCCESS)
+            var offsetXStatus = ((await Gvcp.RegistersDictionary[nameof(RegisterName.OffsetX)].SetValue(offsetX).ConfigureAwait(false)) as GvcpReply).Status == GvcpStatus.GEV_STATUS_SUCCESS;
+            var offsetYStatus = ((await Gvcp.RegistersDictionary[nameof(RegisterName.OffsetY)].SetValue(offsetY).ConfigureAwait(false)) as GvcpReply).Status == GvcpStatus.GEV_STATUS_SUCCESS;
+            if (offsetYStatus && offsetYStatus)
             {
-                OffsetX = reply.RegisterValues[0];
-                OffsetY = reply.RegisterValues[1];
+                OffsetX = (uint)await Gvcp.RegistersDictionary[nameof(RegisterName.OffsetX)].GetValue().ConfigureAwait(false);
+                OffsetY = (uint)await Gvcp.RegistersDictionary[nameof(RegisterName.OffsetY)].GetValue().ConfigureAwait(false);
             }
+
```

Comment:
you can use the multi write register command by fetching the address value first from the register dictionary then use the multi write command